hardware/detect.py

```diff
@@ -293,8 +295,15 @@ def detect_disks(hw_lst):
 
         diskinfo.get_disk_id(name, hw_lst)
 
-        # TODO(rpittau): add smart-tools support for nvme devices
-        if not name.startswith('nvme'):
+        # smartctl support
+        if not which("smartctl"):
+            sys.stderr.write("Cannot find smartctl, exiting\n")
+            return
```

Comment:
I'm a bit worried about this `return`, since if somebody adds more code to this function, it will not work if `smartctl` is not found.

Reply:
mmm right, I guess it needs to be wrapped in a different way, only the smart parsing code should depend by this